hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -175,13 +179,17 @@ public CacheMergePolicyProvider getMergePolicyProvider() {
     }
 
     public Object getMergePolicy(String name) {
-        CacheConfig cacheConfig = configs.get(name);
+        CacheConfig cacheConfig = getCacheConfig(name);
         String mergePolicyName = cacheConfig.getMergePolicy();
         return mergePolicyProvider.getMergePolicy(mergePolicyName);
     }
 
     public ConcurrentMap<String, CacheConfig> getConfigs() {
-        return configs;
+        ConcurrentMap<String, CacheConfig> cacheConfigs = MapUtil.createConcurrentHashMap(configs.size());
+        for (Map.Entry<String, CacheConfigFuture> config : configs.entrySet()) {
+            cacheConfigs.put(config.getKey(), config.getValue().join());
+        }
+        return cacheConfigs;
```

Comment:
I wonder if this method is still needed. We return a static `ConcurrentMap` that is not mutated by any thread anymore. The only place calling this method is `CacheMergeRunnable`, which uses a static view after this change. It feels it should use `getCacheConfig` instead.

Reply:
@vbekiaris How do you see this point? Can't this static map be an issue?

Reply:
As discussed in private, this is not an issue since the callers work with snapshot copies of the `configs` map.